azure-quantum/azure/quantum/optimization/term.py

```diff
@@ -149,11 +194,141 @@ def reduce_by_variable_state(
 
         return Term(indices=new_ids, c=new_c)
 
-    def __repr__(self):
-        return str(self.__dict__)
 
-    def __eq__(self, other):
-        if isinstance(other, self.__class__):
-            return self.__dict__ == other.__dict__
+class GroupType(str, Enum):
+    combination = "na"
```

Comment:
if I understand, a GroupTerm with type "na" is the same as a regular list of terms? So we are introducing another way to add regular terms?

Reply:
I think it's mostly harmless to add this, but it's more functionality that we didn't need.

Reply:
Keeping the `combination` type but re-routing the interface of adding a grouped combination of terms to an ungrouped list of monomials as we discussed